\chapter{Drug Target Discovery}

\section{Introduction}
  \subsection{Motivation}
    Traditionally, drug discovery involves five steps: target identification, target validation, lead identification, lead optimization and introduction of the new drugs to the public \cite{phoebe2008identifying}. Nevertheless, the speed of new drug development has been slower than anticipated, despite increasing investment \cite{pammolli2011productivity}. It is estimated that the cost of developing a new drug is ~USD 2.6 billion \cite{van1998socio}. One of the main reasons for the enormous cost of drug discovery is due to the high failure rate. 

    Success of drug development largely depends on the validity of targets. However, the majority of drugs fail to complete the development process due to lack of efficacy, and this is often due to the wrong target being pursued \cite{shih2018drug}. Traditionally, drug targets are often identified from hypothesis-driven pre-clinical models, yet preclinical models may not always translate well to clinical applications. For some diseases such as psychiatric disorders, current animal or cell models are still far from capturing the complexity of the human disorder \cite{nestler2010animal}. In addition, some have hypothesized the hypothesis-driven nature of many studies may have led to "filtering" of findings and publication bias, exacerbating the reliability and reproducibility issues of some research findings (cite). On the other hand, the recent decade has witnessed a remarkable growth in “omics” and other forms of big data. As increasing amount of biomedical data has been made available, computational methods can offer a fast, cost-effective and unbiased way to prioritize promising drug targets. Given the limitation of current approaches and the urgent need to develop therapies for diseases, addressing the problem of target identification and drug development from different angles is essential. We believe that computational and experimental approaches can complement each other to improve the efficiency and reliability of drug target finding. 

    It's reported that cardiovascular disease (CVD) currently accounts for nearly
    half of noncommunicable diseases (NCDs), CVD remains the leading global cause of
    death, accounting for 17.3 million deaths per year, and the number is expected to grow to 23.6 million by 2030 \cite{laslett2012worldwide}. Hypertension has a higher prevalence
    than other CVD conditions, and it is the most expensive component of CVD \cite{heidenreich2011forecasting}. In 2015, an estimated 5.2 million deaths globally were caused by DM. \cite{mozaffarian2015executive}. Rheumatoid arthritis (RA) a chronic and progressive autoimmune disease, which leads functional disability, pain and joint destruction, and it is estimated to affect between 0.5 and 1.0\% of the adults globally, and its prevalence increases with age, and females are more likely to affect than males \cite{kvien2004epidemiology}. Psychiatric disorders are another kind of disease that need novel treatments, and development of new therapies is also limited by the difficulty of animal models to fully mimic human psychiatric conditions \cite{powell2017transcriptomic}. Whereas spending on industry-wide research and development has nearly doubled over the past decade to \$45 billion a year, FDA has approved fewer and fewer drugs overall \cite{wilson2011drug}. There is an urgent need for innovative approach to improve development of new medications. Therefore, our study mainly focuses on identifying promising targets for diseases: diabetes mellitus (DM), hypertension (HT), rheumatoid arthritis (RA), and some psychiatric disorders.
  
    In the study, we present a flexible novel computational target discovery framework, in which various machine learning methods can be adopted. It's a data-driven approach to prioritize drug targets for diseases, so it's independent from all other kinds of evidences in Open Targets \cite{koscielny2017open}. Specifically, we employ our model to drug-induced expression profiles with indication as outcome variable to learn the pattern of expression characteristics contributing to treatment potential, and applied the fitted models to transcriptome data derived from gene perturbations (i.e. over-expression or knock-down of specific genes) to predict "treatment potential". We could then prioritize drug targets based on the predicted probabilities from the ML model, which reflects treatment potential. Intuitively, for example over-expression (OE) of gene X leads to an expression profile similar to that of five other drugs that are known to treat diabetes. Then an agonist targeted at X (or other drugs that activate X and related pathways) may also be useful for treating diabetes. 

    In this case we expect the ML model (trained on drugs but applied to gene perturbation data) would output a high predicted probability for gene X, and it can be prioritized for further studies. Let’s consider an opposite scenario in which over-expression of gene Y increases the disease risk. In this case we may observe a lower-than-expected predicted probability of ‘treatment potential’ from the ML model. In the other hand, if we knock down the gene Y, we may obtain a predicted probability higher than expected, but in this situation we cannot claim that gene Y can be served as good drug target, since the mechanism and effect size of gene Y on the disease is unclear. Our approach aims to prioritize drug target candidates with the worth of further studies, and we emphasis more on the potential of drug target discovery of our computational framework, since high quality gene expression data that meets characteristics of our approach is still limited (cite).

  \subsection{Related Works}
    With a rapid rise in the availability of biomedical data and in advancement of artificial intelligence (AI), researchers pay more and more attention to application of computational approach to biomedical areas. Recently, we have seen a rise of research interest in computational target discovery. 
    
    G. Kandoi et al. briefly reviewed applications of machine learning and system biology on the discovery of target proteins \cite{kandoi2015prediction}. In these applications, different kinds of biological properties have been explored using machine learning methods to identify druggable targets \cite{bakheet2009properties, fauman2011structure, li2015large,kumari2015identification, li2007prediction}. A sequence-based prediction method was proposed to identify drug target proteins based on biological features like amino acid composition, and a comprehensive comparison of several machine learning methods was conducted \cite{kumari2015identification}. In another study \cite{bakheet2009properties}, eight key properties of human drug target were summarized, and support vector machine (SVM) was employed to build a classifier on these properties to predict  probabilities of potential targets. In a similar study, the authors extracted physicochemical properties from known drug targets, trained a classifier with these properties, and listed possible drug targets by predicted probabilities from the classifier \cite{li2007prediction}. Network based methods also were employed to identify potential drug targets using topological features of human protein–protein interaction network \cite{li2015large}. These studies aim to discover new targets by making use of structural attributes, but gene-disease association data such as gene expression profiles is informative to identify target genes \cite{emig2013drug, ferrero2017silico, sawada2018predicting, costa2010machine}. In a recent study, gene-disease association data from Open Targets was explored by employing four different machine learning methods, including deep neuron networks, to find novel targets, and it’s reported that a large proportion of new targets identified were supported by previous literatures \cite{ferrero2017silico}. Dorothea Emig et. al. proposed an integrated network-based method to predict drug targets based on disease gene expression profiles and a high-quality interaction network, and some novel drug targets for scleroderma and other types of cancer were presented \cite{emig2013drug}. A most recent study constructed pairwise learning and joint learning methods on chemically and genetically perturbed gene expression profiles, and the value of outcome variable was defined by highly correlated pair given by the direct correlation calculation \cite{sawada2018predicting}. 

    However, our study is different from the previous studies in following aspects. Firstly, we employed four ML machine learning methods and constructed them on drug induced gene expression profiles, indicated by known treatment for the disease. Once the model has been constructed, it’s relatively easy to apply to expression profiles induced by genetic perturbation to explore the association between genes and diseases. Further, our approach is general and flexible, and recently developed ML methods can applied easily. Finally, we carried out in-depth validation on predicted results, such as assessing enrichment for targets provided by Open Targets \cite{koscielny2017open}, which is a platform for systematic drug target identification and prioritization. The platform integrates data from genetics, somatic mutations, expression analysis, drugs, animal models and the literature through robust pipelines analysis and uses a single score to indicate the association of a target with disease \cite{koscielny2017open}.

    However, our study is different from the previous studies, including \cite{sawada2018predicting}, in following aspects. Here we mainly discuss the difference between \cite{sawada2018predicting} and our study. 

    First of all, methodologies and aims between our study and study \cite{sawada2018predicting} are different. Our study used ML methods to assess how the expression profiles from gene perturbations are related to those of drugs. The study mainly relied on Pearson correlation to assess the similarity between transcriptomic changes from gene perturbation and those from drugs. An advantage is that many of these ML methods (e.g. SVM, RF, GBM) can accommodate non-linear and more complex relationships between features and the outcome, and different approaches can learn the relationship in different ways. In addition, they used transcriptomic data from gene perturbations mainly to predict drug-protein interactions (mainly using linear models); here we focused on integrating such data with ML approaches to predict drug targets for specific diseases. In short, our method is more advanced and general, which may capture complex relationship present. 

    Many of the previous works (including \cite{sawada2018predicting, li2015large, emig2013drug}) were based on networks of disease, drugs and proteins. For example, for similar diseases (e.g. schizophrenia and bipolar disorder), they may share similar proteins as drug targets. Network-based analysis is a powerful approach, but are still relatively dependent on similarity between entities, hence less capable of discovering novel drug targets. 

    Validation of drug-disease or drug-target predictions from computational methods has always been a difficult task. As reported by \cite{guney2017reproducible}, a cross-validation approach may over-estimate predictive accuracy, as the training and testing set may have overlapping drugs. Also, drugs that are highly similar may be split into train and test sets, hence the similarity of training and testing set may be higher than anticipated in real-life scenarios. Most of these studies evaluate validity of results using performance evaluation metrics (e.g. AUC-ROC) under the framework of cross-validity, so results of these studies actually are overoptimistic.

    In summary, contributions of our study are stated as below. Firstly, we proposed a general framework for identifying targets based on ML methods using expression profiles. Our approach does not rely on a specific kind of models, and any classification algorithms can be incorporated into the framework. Unlike the previous study \cite{ferrero2017silico}, our approach is independent from Open Targets \cite{koscielny2017open}. Secondly, comparing to hypothesis-driven pre-clinical models, which usually take some time to identify promising drug targets, our approach provides a fast cost effective and systematic way to prioritize drug target candidates for further studies. Thirdly, some state-of-the-art machine learning methods are employed to explore potential drug targets for diseases, including diabetes mellitus (DM), hypertension (HT), schizophrenia (SCZ),  rheumatoid arthritis (RA), and bipolar disorders (BP). Lastly, systematic validation analysis was carried out to validate results of our studies, and previous literatures also supported these candidates. 

\section{Datasets and Methods}
  \subsection{Datasets}
    The gene expression profiles were downloaded from The Library of Integrated Network Based Cellular Signatures (LINCS), containing drug induced gene expression profiles and gene  expression profiles induced by over-expression (OE) or knock down (KD) of specific genes \cite{subramanian2017next}. 
      
    For the consistency of genes included in the study, we removed expression of genes appeared either in drug expression profiles or in genetically perturbed expression profiles one dataset, and kept expression information of genes presented in the both two. After the processing, learning models fitted on drug expression profiles can be applied to KD/OE expression profiles. In experiments, the drug expression profile dataset has 1158 observations, each measured in 7467 genes. OE and KD expression datasets have 2413 and 4326 samples respectively, and both have expression information for 7467 genes.

    Drug indications for drug expression profiles were extracted from Anatomical Therapeutic Chemical (ATC) classification system, managed by the World Health Organization Collaborating Centre for Drug Statistics Methodology (WHOCC), and the MEDication Indication Resource high precision subset (MEDI-HPS) \cite{wei2013development}. MEDI-HPS indication resource includes indications extracted from RxNorm, Side Effect Resource 2, MedlinePlus, and Wikipedia \cite{wei2013development}. The high-precision subset only considers medications indicated by RxNorm or appeared in two of other three resource and has 13,304 unique indications for 2,136 medications \cite{wei2013development}. Indication prevalence information was also provided in a further study \cite{wei2013validation}. 

    The outcome for each drug expression profile was defined as the indication of its corresponding drug for a specific disease, which is obtained by examining whether the drug is known for treat the disease. If a drug is known as a treatment to the disease, the outcome of corresponding expression profile is 1; otherwise the outcome is 0. In this study we explored five kinds of diseases: hypertension (HT), diabetes mellitus (DM), rheumatoid arthritis(RA), bipolar disorder (BP), and schizophrenia (SCZ).
    
  \subsection{Methods}
    Here we proposed a general computational framework for prioritizing drug targets for further study, in which any classification algorithms are applicable, hence new advance in classification algorithms can be integrated into our framework. In this study, state-of-the-art machine learning methods, including support vector machine (SVM) \cite{cortes1995support}, gradient boosting machine (GBM) with trees \cite{friedman2001greedy}, random forest (RF) \cite{breiman2001random} and logistic regression with elastic net penalty (EN) \cite{zou2005regularization}, were employed to show the efficacy of our approach.

    Specifically, Our models are built on drug expression profiles, which are labelled by whether their corresponding drugs are known for treat the disease, and then applied to over-expression (OE) and knock-down (KD) gene expression profiles to yield predicted probabilities. The over-expressed / knock-down gene with high prediction probabilities are potential targets to the disease. 

    Specifically, we trained our ML models on drug expression profiles with outcome defined as whether corresponding drugs can treat the disease, and then applied the fitted model to OE/KD expression profiles to predict the probabilities of "treatment"/"treat potential" by over-expressing or knocking down corresponding genes.

    In this section, we mainly discuss three aspects regarding our approach: tuning hyperparameterm, model evaluation, and external validation.

    \subsubsection{Model Specifications}
      We utilized four machine learning methods mentioned above to learn the pattern of expressions that explains treatments of the disease. As the number of drugs known for treat the disease is small, classes for positive outcomes and for negative outcomes are imbalanced. In practice, the strategy of balanced class weights was adopt, which emphasizes more on the minority class to balance the importance of the positive and negative class,  and our previous study has reported that this strategy can improve the performance of ML models \cite{zhao2018drug}. 
      
      We implemented SVM, RF and GBM models using Python package "scikit-learn" \cite{pedregosa2011scikit}. Following the strategy recommended by the tutorial \cite{hsu2003practical}, we adopted a two-step hyperparameter tuning with gridsearchCV provided by the package \cite{pedregosa2011scikit}. 
      
      Specifically, we first defined a broad hyperparameter grid with much bigger step size along axis of each parameter and performed a gridsearchCV for learning algorithms on training set, and then we refined the parameter grid based on the performance of learning algorithms with different parameter combinations. 
   
      In the study, we considered three hyperparameters for SVM, including the kernel type , regularization parameter C and kernel coefficient $\gamma$. Here we used radial basis function (rbf) as the kernel, and C and $\gamma$ were chosen from (-5, 15) and (-15, 3) in log-2 space respectively. For random forests (RF), the number of features considered for each splitting (\textit{max\_features}) and the minimum number of samples required to be at a leaf node (\textit{min\_samples\_leaf}) were used to restrict the complexity of RF. We fixed the number of tree to 1000, and selected \textit{max\_features} and \textit{min\_samples\_leaf} from \{800, 1000, 1500, 2000, 3000, 5000\} and \{1, 3, 5, 10, 30, 50, 80\} respectively. Unlike RF, gradient boosting machine (GBM) is an assembled methods, but in a given iteration GBM gives more emphasis to observations that are misclassified in previous iterations. The characteristics enable it to optimize an arbitrary differentiable loss function. For GBM, learning rate was chosen from \{0.005, 0.01, 0.015, 0.02, 0.03, 0.05\}, the number of boosting iterations from the sequence from 100 to 1001 with step size 50, maximum depth of each estimator from \{2, 3, 5, 10\} and maximum number of features from \{10, 30, 50, 100, 500, 1000\}. Subsampling proportion was fixed to 1 in order to reduce computation burden. Finally, logistic regression with elastic net regularization (EN) was implemented using the R package "glmnet", with hyperparameter $\alpha$ ranging from 0 to 1 with step size 0.1 and the default setting for $\lambda$. In the model, $\alpha$ regulates the sparsity, and $\lambda$ is response for overall regularization. 
  
      In the study, a nested 5-fold cross validation (CV) was employed to choose the best hyperparameters and evaluate performance for each ML algorithm. Here we iteratively split the dataset into three pieces, named training set, validation set and test set. Learning algorithms were trained on the training set, validation set was used to choose hpyerparameters and the performance of ML models were evaluated on the test set, which is independent from the the dataset for model training and validation. In regard to the potential overestimation of model performance with simple CV, the strategy of nested CV can evaluate model performance more accurately \cite{varma2006bias}. More Specifically, in nested CV, we chose the hyperparameters based on the performance of corresponding models in the inner CV; in the outer CV a holdout test set was used to evaluate predictive performance of the optimal model that was chosen in the inner CV. Note that in the experiment the splitting of datasets in the nested CV is the same for different ML models by setting a same random seed. 
  
    \subsubsection{Performance Evaluation}
      The performance of different ML models was evaluated by log Loss, area under the receiver operating characteristic curve (ROC-AUC) and area under the precision recall curve (PR-AUC). Log loss, related to cross-entropy, computes the negative log-likelihood of the true labels predicted by classification models, and the smaller values of log loss indicates that the model performs better. Unlike log loss, ROC-AUC measures the area under the receiver operating characteristic curve of the plot of true positive rate (TPR) against the false positive rate (FPR). On the other hand, PR-AUC measures the area under the curve of the plot of precision against recall. The study suggests that PR-AUC is more favorable in model evaluation when dataset is imbalanced \cite{davis2006relationship}.
  
    \subsubsection{External Validation}
      Though evaluation metrics can quantitatively assess the predictive performance of our ML models, a more direct approach is to assess whether our approach can rediscover known targets or potential targets verified by others' studies for disease. With this aim, we conducted additional statistical analysis using targets for disease downloaded from Open Targets \cite{koscielny2017open} to validate our results. Note our approach is independent from Open Targets. Practically, we are interested in whether the predicted probabilities of known targets from our ML models are statistically significant different from those of other genes.

      Target information of disease was downloaded from Open Targets for external validation \cite{koscielny2017open}. Open Targets provides a single float valued score ranging from 0 to 1 to indicate the association between targets and disease, so we defined a cutoff to filter off a number of targets considered as relevance to the disease. In practice, we used a sequence of cutoff ranging from 1 to 0 with step size 0.1. For each cutoff, the targets with scores greater than the cutoff were considered as valid targets of disease, and those targets with lower score were grouped into the other class. Next, we conducted one-tailed t-tests to assess that whether genes with high predicted probabilities from Our ML models can enrich the valid targets of disease. Bonferroni correction was used to adjust multiple testing \cite{benjamini1995controlling} .
  
\section{Experiment Results}
  
  \subsection{Model Performance}
    Average predictive performance of different ML models, measured in log loss, AUC-ROC and AUC-PR, is presented in Table \ref{tab:target_ml_performance}. It shows that SVM performed the best cross four datasets in term of log loss, and the performance of RF and GBM were similar, slight worse than that of SVM, but the difference is small.

    \begin{table}[htbp]
      \centering
      \caption{Average predictive performance of different machine learning methods across four datasets}
      \begin{threeparttable}
        \begin{tabular}{ccccc}
        \toprule
              & \multicolumn{1}{l}{ATC DM} & \multicolumn{1}{l}{ATC HT} & \multicolumn{1}{l}
              {MEDI-HPS RA} & \multicolumn{1}{l}{ATC SCZ} \\
        \midrule
              & \multicolumn{4}{c}{\textit{Average Log Loss}} \\
        SVM   & \textbf{0.08}  & \textbf{0.2366} & \textbf{0.1209} & \textbf{0.141} \\
        RF    &       0.0836   &     0.2471      &       0.1261    &     0.1462 \\
        GBM   & 0.0875 & 0.2523 & 0.1308 & 0.1555 \\
        EN    & 0.5752 & 0.6781 & 0.6312 & 0.5114 \\
              & \multicolumn{4}{c}{\textit{Average AUC-ROC}} \\
        SVM   &       0.6232     & \textbf{0.5433} &      0.4972     & \textbf{0.7582} \\
        RF    &       0.6024     &     0.5488      &      0.5706     &      0.7377 \\
        GBM   &       0.5404     &     0.5516      &      0.5244     &      0.7474 \\
        EN    & \textbf{0.6485}  &     0.5506      & \textbf{0.5788} &      0.7496 \\
              & \multicolumn{4}{c}{\textit{Average AUC-PR}} \\
        SVM   & \textbf{0.0834} &     0.0804      &     0.0649      & \textbf{0.2402} \\
        RF    &     0.0616      &     0.0884      &     0.0471      &     0.2113 \\
        GBM   &     0.0578      & \textbf{0.0937} &     0.0485      &     0.2106 \\
        EN    &     0.0338      &     0.0792      & \textbf{0.0706} &     0.2362 \\
        \bottomrule
        \end{tabular}%
        \begin{tablenotes}
          \item 1. The figure for best performance of learning algorithms for each dataset for each evaluation metric is in bold.
          \item 2. MEDI-HPS: MEDication Indication-High Precision Subset; ATC: Anatomical Therapeutic Chemical classification.
          \item 3. Abbreviations: DM stands for diabetes mellitus, HT for hypertension, SCZ for schizophrenia, RA for rheumatoid arthritis, BP for bipolar disorders.          
        \end{tablenotes}
      \end{threeparttable}
      \label{tab:target_ml_performance}%
    \end{table}

    When considering AUC-ROC as performance evaluation metric, we find that both SVM and EN have the best performance in two datasets. Specifically, SVM outperforms other methods in ATC-HT and ATC-SCZ dataset, while EN achieves the best performed in the other two datasets. Notably, all ML models performs much better in ATC-SCZ data than in other three datasets. In term of AUC-PR, the performance of all ML is varied. SVM outperforms other methods in ATC DM and ATC SCZ datasets, but GBM and EN have the best performance in other two datasets.

  \subsection{External Validation}  
    \begin{table}[htbp]
      \centering
      \caption{enrichment for target genes of HT by results on ATC-HT dataset}
      \begin{threeparttable}
        \tabcolsep=0.10cm
        \begin{tabular}{ccccccccc}
          \toprule
                & SVM   & RF    & \multicolumn{2}{c}{RF} & \multicolumn{2}{c}{GBM} & \multicolumn{2}{c}{EN} \\
          thresholds & \textit{P-value} & \textit{q-value} & \textit{P-value} & \textit{q-value} & \textit{P-value} & \textit{q-value} & \textit{P-value} & \textit{q-value} \\
          \midrule
          1     & 4.81E-03 & 5.77E-03 & 3.31E-02 & 3.97E-02 & 9.81E-02 & 1.18E-01 & 2.09E-02 & \textbf{2.60E-02} \\
          0.8   & 4.32E-03 & 5.77E-03 & 2.56E-02 & 3.84E-02 & 8.29E-02 & 1.18E-01 & 1.61E-02 & \textbf{2.60E-02} \\
          0.6   & \textbf{4.41E-04} & \textbf{2.48E-03} & 4.94E-03 & \textbf{1.48E-02} & 1.76E-02 & \textbf{5.28E-02} & \textbf{5.68E-03} & \textbf{2.60E-02} \\
          0.4   & 8.26E-04 & 2.48E-03 & \textbf{4.86E-03} & \textbf{1.48E-02} & \textbf{1.60E-02} & 5.28E-02 & 1.12E-02 & \textbf{2.60E-02} \\
          0.2   & 2.04E-03 & 4.08E-03 & 1.19E-02 & 2.38E-02 & 2.91E-02 & 5.82E-02 & 2.17E-02 & \textbf{2.60E-02} \\
          0     & 1.56E-01 & 1.56E-01 & 6.84E-01 & 6.84E-01 & 1.96E-01 & 1.96E-01 & 1.12E-01 & 1.12E-01 \\
          \bottomrule
          \end{tabular}%
          \begin{tablenotes}
            \item 1. Figures in the table are p-values calculated by two tailed t-test with alternative hypothesis that the mean probability of genes matched by Open Targets is different than that of genes unmatched. 
            \item 2. The lowest p-values/q-values for every ML model in each dataset are in bold.
            \item 3. Abbreviations are defined the same as Table \ref{tab:target_ml_performance}.  
          \end{tablenotes}
        \end{threeparttable}
        \label{tab:repurposing_enrichment_ht}%
      \end{table}%

      \begin{table}[htbp]
        \centering
        \caption{enrichment for target genes of DM by results on ATC-DM dataset}
        \begin{threeparttable}
          \tabcolsep=0.10cm
          \begin{tabular}{ccccccccc}
            \toprule
                  & SVM   & RF    & \multicolumn{2}{c}{RF} & \multicolumn{2}{c}{GBM} & \multicolumn{2}{c}{EN} \\
            thresholds & \textit{P-value} & \textit{q-value} & \textit{P-value} & \textit{q-value} & \textit{P-value} & \textit{q-value} & \textit{P-value} & \textit{q-value} \\
            \midrule
            1     & 5.78E-02 & 8.67E-02 & 6.53E-02 & 9.80E-02 & 3.28E-05 & 8.62E-05 & 2.32E-03 & 4.64E-03 \\
            0.8   & 2.33E-02 & 4.66E-02 & 2.13E-02 & 6.39E-02 & 4.31E-05 & 8.62E-05 & 1.67E-03 & 4.64E-03 \\
            0.6   & \textbf{1.26E-02} & \textbf{4.66E-02} & \textbf{1.37E-02} & \textbf{6.39E-02} & \textbf{1.61E-05} & \textbf{8.62E-05} & \textbf{1.57E-03} & \textbf{4.64E-03} \\
            0.4   & 2.32E-02 & 4.66E-02 & 4.32E-02 & 8.64E-02 & 1.72E-03 & 2.58E-03 & 5.22E-03 & 7.83E-03 \\
            0.2   & 6.43E-01 & 6.43E-01 & 3.67E-01 & 4.40E-01 & 1.71E-02 & 2.05E-02 & 3.55E-02 & 4.26E-02 \\
            0     & 3.00E-01 & 3.60E-01 & 6.24E-01 & 6.24E-01 & 8.10E-01 & 8.10E-01 & 8.21E-02 & 8.21E-02 \\
            \bottomrule
            \end{tabular}%
            \begin{tablenotes}
              \item 1. Illustrations of the table are the same as Table \ref{tab:repurposing_enrichment_ht}
            \end{tablenotes}
          \end{threeparttable}
          \label{tab:repurposing_enrichment_dm}%
        \end{table}%

        \begin{table}[htbp]
          \centering
          \caption{enrichment for target genes of RA by results on MEDI-HPS RA dataset}
          \begin{threeparttable}
            \tabcolsep=0.10cm
            \begin{tabular}{ccccccccc}
              \toprule
                  & SVM   & RF    & \multicolumn{2}{c}{RF} & \multicolumn{2}{c}{GBM} & \multicolumn{2}{c}{EN} \\
            thresholds & \textit{P-value} & \textit{q-value} & \textit{P-value} & \textit{q-value} & \textit{P-value} & \textit{q-value} & \textit{P-value} & \textit{q-value} \\
            \midrule
            1     & 1.18E-01 & 2.72E-01 & 6.23E-04 & 1.87E-03 & 2.01E-02 & 4.50E-02 & 9.22E-01 & 9.94E-01 \\
            0.8   & 1.36E-01 & \textbf{2.72E-01} & \textbf{3.93E-04} & \textbf{1.87E-03} & \textbf{8.53E-03} & \textbf{4.50E-02} & 9.94E-01 & 9.94E-01 \\
            0.6   & \textbf{1.18E-01} & 2.72E-01 & 1.41E-01 & 1.69E-01 & \textbf{3.67E-01} & 3.67E-01 & 9.20E-01 & 9.94E-01 \\
            0.4   & 6.44E-01 & 6.44E-01 & 1.14E-02 & 2.28E-02 & 2.25E-02 & 4.50E-02 & 2.69E-01 & 5.38E-01 \\
            0.2   & 3.12E-01 & 4.45E-01 & 8.47E-02 & 1.27E-01 & 4.15E-02 & 6.23E-02 & 8.48E-02 & \textbf{5.09E-01} \\
            0     & 3.71E-01 & 4.45E-01 & 7.01E-01 & 7.01E-01 & 1.96E-01 & 2.35E-01 & \textbf{2.56E-01} & 5.38E-01 \\
            \bottomrule
            \end{tabular}%
            \begin{tablenotes}
              \item 1. Illustrations of the table are the same as Table \ref{tab:repurposing_enrichment_ht}
            \end{tablenotes}
          \end{threeparttable}
          \label{tab:repurposing_enrichment_ra}%
        \end{table}%

        \begin{table}[htbp]
          \centering
          \caption{enrichment for target genes of DM by results on ATC SCZ dataset}
          \begin{threeparttable}
            \tabcolsep=0.10cm
            \begin{tabular}{ccccccccc}
            \toprule
                  & SVM   & RF    & \multicolumn{2}{c}{RF} & \multicolumn{2}{c}{GBM} & \multicolumn{2}{c}{EN} \\
            thresholds & \textit{P-value} & \textit{q-value} & \textit{P-value} & \textit{q-value} & \textit{P-value} & \textit{q-value} & \textit{P-value} & \textit{q-value} \\
            \midrule
            1     & 3.32E-01 & 4.98E-01 & 2.84E-01 & 5.68E-01 & \textbf{2.57E-01} & \textbf{7.92E-01} & 3.56E-01 & \textbf{5.76E-01} \\
            0.8   & 4.66E-01 & 5.59E-01 & 2.47E-01 & \textbf{5.68E-01} & 2.64E-01 & 7.92E-01 & \textbf{2.80E-01} & 5.76E-01 \\
            0.6   & 2.18E-02 & 6.54E-02 & 7.78E-01 & 8.97E-01 & 9.94E-01 & 9.94E-01 & 7.47E-01 & 7.47E-01 \\
            0.4   & \textbf{1.91E-02} & 6.54E-02 & 8.62E-01 & 8.97E-01 & 7.97E-01 & 9.94E-01 & 3.84E-01 & 5.76E-01 \\
            0.2   & 7.00E-02 & \textbf{1.40E-01} & 8.97E-01 & 8.97E-01 & 5.42E-01 & 9.94E-01 & 7.18E-01 & 7.47E-01 \\
            0     & 7.11E-01 & 7.11E-01 & \textbf{1.85E-01} & 5.68E-01 & 9.01E-01 & 9.94E-01 & 3.14E-01 & 5.76E-01 \\
            \bottomrule
            \end{tabular}%
            \begin{tablenotes}
              \item 1. Illustrations of the table are the same as Table \ref{tab:repurposing_enrichment_ht}
            \end{tablenotes}
          \end{threeparttable}
          \label{tab:repurposing_enrichment_scz}%
        \end{table}%


        \begin{table}[htbp]
          \centering
          \caption{enrichment for target genes of BP by results on ATC SCZ dataset}
          \begin{threeparttable}
            \tabcolsep=0.10cm
            \begin{tabular}{ccccccccc}
            \toprule
                  & SVM   & RF    & \multicolumn{2}{c}{RF} & \multicolumn{2}{c}{GBM} & \multicolumn{2}{c}{EN} \\
            thresholds & \textit{P-value} & \textit{q-value} & \textit{P-value} & \textit{q-value} & \textit{P-value} & \textit{q-value} & \textit{P-value} & \textit{q-value} \\
            \midrule
            1     & 4.14E-01 & 6.99E-01 & 7.24E-01 & 7.97E-01 & 5.88E-01 & 7.06E-01 & 5.59E-01 & 6.71E-01 \\
            0.8   & 4.66E-01 & 6.99E-01 & 7.58E-01 & \textbf{7.97E-01} & 9.43E-01 & 9.43E-01 & 9.43E-01 & 9.43E-01 \\
            0.6   & 8.57E-01 & 8.57E-01 & 6.84E-01 & 7.97E-01 & 2.56E-01 & 3.84E-01 & 2.90E-02 & 5.80E-02 \\
            0.4   & 8.57E-01 & 8.57E-01 & 6.84E-01 & 7.97E-01 & 2.56E-01 & 3.84E-01 & 2.90E-02 & 5.80E-02 \\
            0.2   & 4.13E-01 & 6.99E-01 & \textbf{3.17E-01} & 7.97E-01 & \textbf{6.03E-02} & \textbf{3.62E-01} & \textbf{1.45E-03} & \textbf{8.70E-03} \\
            0     & \textbf{1.31E-02} & \textbf{7.86E-02} & 7.97E-01 & 7.97E-01 & 1.91E-01 & 3.84E-01 & 4.76E-01 & 6.71E-01 \\
            \midrule
            \end{tabular}%
            \begin{tablenotes}
              \item 1. Illustrations of the table are the same as Table \ref{tab:repurposing_enrichment_ht}
            \end{tablenotes}
          \end{threeparttable}
          \label{tab:repurposing_enrichment_bp}%
        \end{table}%

    We conducted enrichment test to validate whether genes ranked by predicted probabilities from ML models are enriched for targets extracted from Open Targets. The results of enrichment test for over-expressed genes for different diseases are shown in Tables \ref{tab:repurposing_enrichment_ht}, \ref{tab:repurposing_enrichment_dm}, \ref{tab:repurposing_enrichment_ra}, \ref{tab:repurposing_enrichment_scz}, \ref{tab:repurposing_enrichment_bp}. There is no statistical significance showed in enrichment tests for knockdown genes, and we have placed the results in supplementary for readers of interest. 
    
    From tables \ref{tab:repurposing_enrichment_ht}, \ref{tab:repurposing_enrichment_dm}, \ref{tab:repurposing_enrichment_ra}, we observed statistical significance for all ML models. This indicates that our methods well rediscovers targets provided by Open Targets. Further, statistical significances for every model in each dataset were achieved at thresholds in the middle range. In table \ref{tab:repurposing_enrichment_scz}, \ref{tab:repurposing_enrichment_bp} we also see several significant p-values, providing some evidences to validate our results.

  \subsection{Literature Support}
    Our model generated several potential druggable targets for different diseases, and we are interested in whether these potential targets have been studies in previous literatures. Here we have done a detailed literature support analysis for potential targets from approach for different disease to support our findings. 

    Top potential drug target candidates suggested by our model for DM include SGK1, ESR1, CISH, MAPK4K4, UGCG. Our model suggested antagonizing SGK1 will likely have therapeutic effect on DM. In diabetic animal models, SGK1 expressions were up-regulated \cite{hills2006high, xuebin2005expression,chang2007enhancement}, the upregulation likely to be induced by production of advanced glycation products (AGE) \cite{hills2006high,chang2007enhancement}. SGK1 is critical to the development of diabetic neuropathy, and inhibition of SGK1 by fluvastatin has shown favorable hindrance in the pathophysiology progression of DM \cite{xuebin2005expression}. The predicted probability for ESR1 indicates that inhibiting the estrogen receptor will very likely benefit DM treatment, and this postulation is also supported by evidence in human \cite{barros2006estrogen} and animal \cite{weigt2015effects} studies. The mechanistic linkage lies in the modulating effects of estrogen receptors on genes related to insulin sensitivity and glucose reuptake \cite{barros2006estrogen}. Specifically, in studies of the mouse who cannot synthesize estrogen (ArKO), the mice showed increased adipose fat deposit as well as reduced insulin sensitivity \cite{jones2000aromatase,takeda2003progressive}. CISH (Cytokine-inducible SH2) containing Cish protein is involved in the signaling pathway of hGC, and Induction of this pathway by hGC minigene has been shown to promote beta-islet cell proliferation in murine model \cite{baan2015transgenic}. MAPK4K4 is also an antagonized drug target suggested by our model for DM. In a cell culture experiment, inhibition of MAPK4K4 by RNA interference was able to completely reverse the insulin resisting effect of TNF-alpha \cite{bouzakri2007map4k4}. UGCG (Ceramide and its metabolites) is known as an inhibitor of insulin sensitivity as well as has demonstrated its inhibitory effect of downstream component phosphorylation. Aerts et al. demonstrated that both ceramide and its downstream metabolites could reduce insulin sensitivity in vivo and that treatment with AMP-DNM could reverse the insulin insensitivity \cite{aerts2007pharmacological}. 
  
    Our approach also identified several promising targets for rheumatoid arthritis. Traditionally, substance P inhibition has an important implication in the pathophysiology of RA \cite{lisowska2015substance,garrett1992role,green2005gastrin,keeble2004role,lotz1987substance,okamura2017dual,lam1990mediators,lam1991neurogenic}. It serves as a pain transmitter, exacerbates the inflammatory process in arthritic joints and worsens disease progression. Our study identifies SP inhibition as a favorable drug target, and this has already been proven decades ago with capsasin administration \cite{lam1989capsaicin} as well as improving dexamethasone \cite{lam2010substance} efficacy when co-administered. CTNNBIP-1, which encodes for $\beta$-catenin, is identified as an very promising effective therapeutic target by our RF model, and it has already been studied extensively. $\beta$-catenin is a critical player of the Wnt signaling pathway that initiates arthritic progression and development in joints \cite{sen2005wnt,zhou2017wnt}, and studies considered animals with dysfunction of this pathway as a disease model for RA \cite{wu2010beta,zhou2017wnt}. There is a pool of potential pharmacological interventions that have been shown to thwart, improve or even repair disease condition at arthritic joints, including herbal isolate Artemisinin \cite{zhong2018artemisinin}, small molecular inhibitors \cite{lietman2018inhibition,landman2013small,landman2013small,dell2017pharmacological}, FDA approved SSRI \cite{miyamoto2017fluoxetine}, and even naturally occurring polyphenol from diet \cite{li2018resveratrol}. LMP7 is subunit of the immunoproteosome important for generating peptide fragments on the MHC class I receptor \cite{joeris2012proteasome}, thus implicating the role it plays in generating autoimmune response in diseases like RA. Basler et al. found that LMP7 alone was not sufficient, and that LMP2 must also be co-inhibited in order to block immunity \cite{basler2018co}. These findings suggest that ONX 0914, a selective inhibitor for LMP7 used for other autoimmune diseases \cite{althof2018immunoproteasome,liu2017onx,verbrugge2012targeting}, may be repositioned to treat RA. Inhibition of LMP7 is also agreed with our findings. Our model suggests that antagonizing NR4A2 could help treat RA, which is consistent with physiological findings. As NR4A2 is an orphan nuclear receptor that is responsible for proinflammatory responses particularly to IL-8 in RA \cite{aherne2009identification}. It was also found that it plays as a downstream response elements of TNF-alpha \cite{mix2012orphan} as well as transcription factor for immunomodulatory peptide hormone prolactin \cite{mccoy2015orphan}. Currently, much of the DMARD in clinical use were developed against TNF. Our SVM model shows this receptor may bring about therapeutic effects. It is not as strong as the drug targets mentioned above, and it's not the most efficacious biologic antibody drug by evidence from applying TNF monotherapy studies\cite{aletaha2018diagnosis} .
  
    Known popular targets discovered by our approach for hypertension has antagonizing angiotensin II and glucocorticoid. Our model strongly suggests antagonizing angiotensin II and glucocorticoid to be targeted as a therapeutic approach to hypertension, while atrial natriuretic peptide should be agonized to treat hypertension. These two targets need no justification themselves, but they proves the physiological relevance and validity of our model and also demonstrates that not only our approach can produce novel drug targets, but some of our results are compatible to traditional pharmacological interventions. Studies show that cAMP Response Element Binding (CREB) protein participates in the inflammatory, vascular remodeling and apoptotic processes \cite{ichiki2006role}. Increasing CREB activity with phosphodiesterase inhibitors may be considered as a novel target in treating pulmonary hypertension, but cAMP signaling appears to be positively regulating calcium entry in vascular smooth muscles \cite{pulver2004store}. Interestingly, our model learned pattern of the biological mechanisms for decreasing or increasing blood pressure. For instance, our model strongly recommends that inhibiting corticosteroid-binding protein will reduce blood pressure, and its biological effect is clear, since SERPINA6 deficiency usually leads low blood pressure \cite{torpy2001familial}. 
  
    Our method discovered multiple potential targets for SCZ that are also appeared in GWAS publications \cite{zhang2015functional,golimbet2014study,lee2013pathway,sinclair2012glucocorticoid,ahmad2015association,kishi2011sirt1,athanasiou2011candidate,hoenicka2010sexually,tein2008short,sun2004cldn5,chen2004case,yu2008association,hashimoto2005functional}, independently supporting the validity of our approach. Most importantly, an advantage of our method is that predicted probabilities from our approach can indicate the direction of treatment of potential targets on disease instead of a simply association between alleles and disease. We also observed gene candidates with biological evidence in the pathophsyiology of SCZ. PIP5K2A acts as the agonist of amino acid transporter EAAT3, which helps glutamate reuptake. Without two functioning copies of PIP5K2A, there will be an accumulation of excitatory neurotransmitter, causing neurotoxicity \cite{fedorenko2009pip5k2a}. Our finding suggests the direction of possible treatment approach is to agonize this target. The dopaminergic pathways in the brain play crucial roles in the explanation of the pathophysiology of SCZ. Our model proposes that a potential treatment approach is to agonize the D1 receptor. Despite the dominant role of D2 in the nigralstratal pathway afferents makes it the primary target of antipsychootic drugs \cite{howes2009dopamine}, there is an emergent study of the role of D1 with respect to its contribution to the negative symptoms \cite{goldman2004targeting}. Non-human primates models demonstrate that administration of D1 selective agonist enhance spatial working memory, but it is not the case for D2 agonist \cite{goldman2004targeting}. Another possible drug candidate to deal with negative symptoms of SCZ is the 5-HT2C receptor agonist. Although the first line of treatment for SCZ is usually atypical antipsychotic drugs (APD) which will agonize 5-HT2A, pharmacological findings on 5-HT2C drugs has shown promising effects in animal models \cite{floresco2009neural,hemrick2002comparison,pogorelov20175}. It is suggested that activation of this receptor will lead to a selective suppression of DA release in VTA and Nac but not the nigralstraital DA neuron firing \cite{meltzer20115}.
  
    Like SCZ, our approach discovered several drug target candidates for Bipolar Disorder (BP) that are associated gene polymorphism \cite{lee2013pathway,zhang2015functional,golimbet2014study}. Our findings indicate that a potential approach to treat BP is to activate the JUN (a subunit of AP-1) gene. In fact, research has show that the effect is caused by lithium and valporic acids, and specifically AP-1 can be increased by lithium in human neuroblastoma and rat brain tissues and a similar effect of valporic acid has also been observed \cite{yuan1998lithium,asghari1998differential,ozaki1997lithium}. The beneficial effect of AP-1 is due to its neurotrophic and neuroprotective effects \cite{machado2009role}. Previous studies suggests that TH has no association with BIP \cite{grassi1996no,turecki1997lack}. However, later studies show that BIP is associated with an increased amount of tyrosine hydroxylase, and the chronic treatment of valporic acid has been shown to reduce the mRNA level of TH \cite{pantazopoulos2004differences,muglia2002dopamine}. TH gene is shown to associate with the depressive phase of the patient \cite{muglia2002dopamine}. Our findings correctly identify its direction of effect and propose it as potential treatment regime of antagonizing TH.

\section{Discussion}
  In this study, we presented a novel computational approach to identify promising drug targets by incorporating gene expression profiles. This approach is general as it can widely incorporate any classification algorithms. Four state-of-the art machine learning methods were used to demonstrate the potential of our approach, and their performances were compared by three different evaluation metrics. Results of enrichment test shows that top genes from our models are enriched for targets from Open Targets for diabetes mellitus, hypertension, schizophrenia, bipolar disorders, and rheumatoid arthritis. Some of top potential drug targets identified by our approach are also supported by previous studies. 

  As far as we know, this work is the first one to directly employ state-of-the-art machine learning methods on both drug induced and genetically perturbed expression data to discover potential drug targets and to shed light on molecular association between genes and diseases. 

  The study adopted four ML methods, which are capable of learning either linear or nonlinear relationship present in data. Because the size of our dataset is p $\gg$ n, logistic regression with elastic net penalty (EN) can automatically select features that contribute the most to prediction and also consider the characteristics of gene expression data, in which there are groups of highly correlated variables, in feature selection. In our case, the performance of EN is quite reasonable, even though it only models linear relationship. SVM uses kernel tricks to map the feature into high dimension feature space, in which our data can readily been separated, and this makes SVM a better choice, even though it is not the best one cross all datasets in term of AUC-ROC and AUC-PR. On the other hand, random forests (RF) and gradient boosting machine (GBM) are well-known tree-based methods. The basic idea behind them is to assemble a number of low correlated weak learners of tree structure to make accurate predictions. They are robust and low variant in new dataset, and interpretable. A main difference between the two is that GBM is a sequential tree model, which adjusts the importance of observations in learning according to the performance of previous fitted trees. Due to the common characteristics of RF and GBM, their performances are comparable overall cross four datasets. In this study, we do not emphasis on particular algorithms, since we are interested in demonstrating the potential of approach in prioritizing drug target candidates and shedding light on their possible directions of mechanism in drug development. Even though many targets listed are supported by previous studies, further validation with well-designed animal models is also necessary before these target candidates are used in drug development. It should be noted that detailed animal experimental validation of one or two candidate targets provides little evidence on whether our approach works well in practice, because there may be chance findings and only a proportion of genes are covered in the study. In addition, the quality of expression data used in the study is limited, and this may affect the result of the study and lead to invalid drug target candidates.
  
  Concerning results of enrichment test, A two sided t-test was carried out for five different diseases, and we see obvious statistical significance for all methods cross five datasets except schizophrenia dataset. This provides an independent evidence to validate our approach.

  Our approach is general and highly flexible. However, further improvement can be made in the following aspects. Firstly, the quality of knock-down expression profiles isn’t as well as over-expression expression profiles, thus leading that no significance is detected in all knock out datasets. Thus, high quality knock-down expression dataset may offer benefits in two ways: 1) this may provide mutual validation between results from knock-down and over-expressed dataset. Specifically, if over-expression of specific gene lead to disease, then knock-down of the gene may induce treatment effect; 2) different targets may be identified through knock-down expression dataset, because different genes works in various different ways to lead to diseases and molecular mechanisms of genes are different cross diseases. 

  Another concern of this study is that our dataset is highly imbalanced, due to insufficient positive observations. In order to address this issue we increased class weight of the minority group, but there are also other strategies to address issue such as SMOTE (Synthetic Minority Over-sampling Technique) \cite{chawla2002smote}, but whether strategies like SMOTE can address this issue in high dimensional setting is still unclear. Thus, we may leave this topic for further investigation. Further, we may also resort to more advanced or recently developed machine learning methods to find more meaningful targets. The scale of the dataset is another issue, and further exploration of larger gene expression dataset is of great interest to us. 

\section{Conclusion}
  This study presented a general computational framework to prioritize drug targets for various diseases. Under the framework, different kinds of machine learning methods can be utilized. The characteristics of our approach demonstrates its generality and flexibility. We applied four state-of-the-art machine learning methods to identify potential drug target of four different diseases, including hypertension and diabetes mellitus, which affect the life of billions people. External validation shows that top candidates of drug target are enriched for targets of different diseases from Open Targets, an independent resource of drug target. Meanwhile, some top target genes were also supported by previous studies.

  Finding promising drug targets for diseases is crucial to drug development, which may shorten the cycle of drug development and contribute to develop drugs with novel mechanisms of actions. The traditional approach using animal models to identify ideal drug targets is time-consuming and suffers a high failure rate, and it's difficult for the traditional approach to find drug targets with novel mechanisms of actions for diseases. On the other hand, we have seen tremendous advance in machine learning methods, especially in deep neuron networks, and a rise in the availability of high quality biomedical data in the past few years. We expect to translate the two resources into medical advances to benefit patients. We hope this study may open a new pathway of discovering new drug targets or pathogenic genes to diseases. The lists of drug targets from the study might provide valuable resources for researchers who are willing to find new drug targets for cardiovascular disease, rheumatoid arthritis or schizophrenia, and to develop new chemicals/compounds to treat these diseases.

\chapterend