\chapter{Drug Target Discovery}

\section{Motivation}
Traditionally, drug discovery involves five steps: target identification, target validation, lead identification, lead optimization and introduction of the new drugs to the public \cite{phoebe2008identifying}. Nevertheless, the speed of new drug development has been slower than anticipated, despite increasing investment \cite{pammolli2011productivity}. It is estimated that the cost of developing a new drug is ~USD 2.6 billion \cite{van1998socio}. One of the main reasons for the enormous cost of drug discovery is due to the high failure rate. 

Success of drug development largely depends on the validity of targets. However, the majority of drugs fail to complete the development process due to lack of efficacy, and this is often due to the wrong target being pursued \cite{shih2018drug}. Traditionally, drug targets are often identified from hypothesis-driven pre-clinical models, yet preclinical models may not always translate well to clinical applications. For some diseases such as psychiatric disorders, current animal or cell models are still far from capturing the complexity of the human disorder (cite). In addition, some have hypothesized the hypothesis-driven nature of many studies may have led to "filtering" of findings and publication bias, exacerbating the reliability and reproducibility issues of some research findings (cite). On the other hand, the recent decade has witnessed a remarkable growth in “omics” and other forms of big data. As increasing amount of biomedical data has been made available, computational methods can offer a fast, cost-effective and unbiased way to prioritize promising drug targets. Given the limitation of current approaches and the urgent need to develop therapies for diseases, addressing the problem of target identification and drug development from different angles is essential. We believe that computational and experimental approaches can complement each other to improve the efficiency and reliability of drug target finding. 
  
In this work, we present a novel computational framework to prioritize drug targets for specific diseases. Briefly, we first fit machine learning (ML) models to predict drug indications from drug-induced expression profiles. The aim is to “learn” expression patterns that are associated with successful treatment of the studied disease. The fitted ML models were then applied to transcriptome data derived from gene perturbations (i.e. over-expression or knock-down of specific genes). We could then prioritize drug targets based on the predicted probabilities from the ML model, which reflects treatment potential. Intuitively, for example over-expression (OE) of gene X leads to an expression profile similar to that of five other drugs that are known to treat diabetes. Then an agonist targeted at X (or other drugs that activate X and related pathways) may also be useful for treating diabetes. 

In this case we expect the ML model (trained on drugs but applied to gene perturbation data) would output a high predicted probability for gene X, and it can be prioritized for further studies. Let’s consider an opposite scenario in which over-expression of gene Y increases the disease risk. In this case we may observe a lower-than-expected predicted probability of ‘treatment potential’ from the ML model. Here we emphasis more on the potential of drug target discovery of our computational framework, since high quality gene expression data that meets characteristics of our approach is still limited (cite). 



\chapterend